### PR TITLE
Added cancel reset tests

### PR DESCRIPTION
**Pull request by RazRegev**

### CODE REVIEW

File: discovery-infra/tests/base_test.py [hunk 1]
```diff
@@ -161,6 +161,14 @@ def wait_for_one_host_to_boot_during_install(cluster_id, api_client):
             cluster_id=cluster_id,
             stages=[consts.HostsProgressStages.REBOOTING])
 
+    @staticmethod
```
Comment (nshidlin) [inline discovery-infra/tests/base_test.py:164]: why not use wait_for_hosts_to_install and pass num_nodes=1?

File: discovery-infra/tests/test_cancel_and_reset.py [hunk 2]
```diff
@@ -181,3 +187,196 @@ def test_cancel_reset_while_disable_workers(self, api_client, node_controller, c
...
+            cluster_id=cluster_id
+        )
+        for i in range(10):
+            logger.debug(
+                'test_cluster_install_and_reset_10_times attempt number: %d',
+                i + 1
+            )
+            self.start_cluster_install(cluster_id, api_client)
+            self.wait_for_installing_in_progress(cluster_id, api_client)
+            assert self.is_cluster_in_cancelled_status(
```
Comment (nshidlin) [inline discovery-infra/tests/test_cancel_and_reset.py:255]: You assert on canceled status without canceling the install

File: discovery-infra/tests/test_cancel_and_reset.py [hunk 3]
```diff
@@ -181,3 +187,197 @@ def test_cancel_reset_while_disable_workers(self, api_client, node_controller, c
...
+
+        self.client.update_params(cluster_id, {
+            'api_vip': str(api_vip),
+            'ingress_vip': str(ingress_vip),
+            #'pull_secret': # add
+        })
+
+        self._install_ready_cluster(api_client, cluster_id)
+
+    def _prepare_cluster_for_installation(
```
Comment (tsorya) [inline discovery-infra/tests/test_cancel_and_reset.py:354]: move it to new cluster object

File: discovery-infra/tests/test_cancel_and_reset.py [hunk 4]
```diff
@@ -181,3 +187,197 @@ def test_cancel_reset_while_disable_workers(self, api_client, node_controller, c
...
+        node_controller,
+        cluster_id
+    ):
+        self.generate_and_download_image(cluster_id, api_client)
+        node_controller.start_all_nodes()
+        self.wait_until_hosts_are_discovered(cluster_id, api_client)
+        self.set_host_roles(cluster_id, api_client)
+        self.set_network_params(cluster_id, api_client, node_controller)
+
+    def _reboot_hosts_and_wait_for_cluster_readiness(
```
Comment (tsorya) [inline discovery-infra/tests/test_cancel_and_reset.py:366]: same

File: discovery-infra/tests/test_cancel_and_reset.py [hunk 5]
```diff
@@ -181,3 +187,197 @@ def test_cancel_reset_while_disable_workers(self, api_client, node_controller, c
...
+    ):
+        self.reboot_required_nodes_into_iso_after_reset(
+            cluster_id=cluster_id,
+            api_client=api_client,
+            controller=node_controller
+        )
+        self.wait_until_hosts_are_discovered(cluster_id, api_client)
+        self.wait_until_cluster_is_ready_for_install(cluster_id, api_client)
+
+    def _install_ready_cluster(self, api_client, cluster_id):
```
Comment (tsorya) [inline discovery-infra/tests/test_cancel_and_reset.py:380]: same

File: discovery-infra/tests/test_cancel_and_reset.py [hunk 6]
```diff
@@ -181,3 +187,197 @@ def test_cancel_reset_while_disable_workers(self, api_client, node_controller, c
...
+        ), f'cluster {cluster_id} failed to reset from ' \
+           f'canceled state'
+        self._reboot_hosts_and_wait_for_cluster_readiness(
+            api_client=api_client,
+            node_controller=node_controller,
+            cluster_id=cluster_id
+        )
+
+        vips = node_controller.get_ingress_and_api_vips()
+        api_vip = IPNetwork(vips['api_vip'])
```
Comment (nshidlin) [inline discovery-infra/tests/test_cancel_and_reset.py:341]: Won't the installation fail if the DNS records aren't updated?

File: discovery-infra/tests/test_cancel_and_reset.py [hunk 7]
```diff
@@ -181,3 +187,197 @@ def test_cancel_reset_while_disable_workers(self, api_client, node_controller, c
...
+            'is %s', cluster_id
+        )
+        self._prepare_cluster_for_installation(
+            api_client=api_client,
+            node_controller=node_controller,
+            cluster_id=cluster_id
+        )
+        self._install_ready_cluster(api_client, cluster_id)
+
+        with pytest.raises(Exception):
```
Comment (nshidlin) [inline discovery-infra/tests/test_cancel_and_reset.py:294]: should be with pytest.raises(ApiException), we know what exception is raised

File: discovery-infra/tests/test_cancel_and_reset.py (hunk 6 above)
Comment (RazRegev) [reply discovery-infra/tests/test_cancel_and_reset.py:293]: Not sure, this test hasn't been completed and currently disabled by default

**Pull request merged**